src/gt4py/next/errors/formatting.py

```diff
@@ -0,0 +1,93 @@
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2023, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import pathlib
+import textwrap
+import traceback
+import types
+from typing import Optional
+
+from gt4py.eve import SourceLocation
+
+
+def get_source_from_location(location: SourceLocation):
+    if not location.filename:
+        raise FileNotFoundError()
+    source_file = pathlib.Path(location.filename)
+    source_code = source_file.read_text()
+    source_lines = source_code.splitlines(False)
+    start_line = location.line
+    end_line = location.end_line + 1 if location.end_line else start_line + 1
+    relevant_lines = source_lines[(start_line - 1) : (end_line - 1)]
+    return "\n".join(relevant_lines)
+
+
+def format_location(loc: SourceLocation, caret: bool = False):
+    filename = loc.filename or "<unknown>"
```

Comment:
Optionally, you could add a docstring here to explain the meaning of the `caret` argument...

Reply:
Done, can you check if it makes things any better? Kinda difficult to explain this in one line. Any suggestions for a better variable name?